### PR TITLE
refactor: cleanup `BalanceSummaryRow` & `BalanceSummarySection`

### LINKED ISSUE #7090: [Task]: Cleanup `BalanceSummaryRow`

### PR DESCRIPTION
**Pull request by evavirseda**
## Summary

Closes #7090 

## Changelog

```
- Cleanup and refactor `BalanceSummaryRow`
- Cleanup `BalanceSummaryRow`
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/BalanceSummarySection.svelte [hunk 1]
```diff
@@ -34,7 +34,7 @@
     <div
         class="w-full flex flex-row flex-grow justify-between space-x-2 {hasChildren ? 'cursor-pointer ' : ''}"
```
Comment (VmMad) [inline packages/shared/components/BalanceSummarySection.svelte:35]:
```suggestion
        class="w-full flex flex-row flex-grow justify-between space-x-2"
        class:cursor-pointer={hasChildren}
```

File: packages/shared/components/BalanceSummarySection.svelte [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
 
-    let expanded = false
+    let expanded: boolean = false
 
     $: hasChildren = !!Object.keys(subBreakdown ?? {}).length
     $: ({ baseCoin } = $selectedAccountAssets?.[$activeProfile?.network?.id] ?? {})
 
     function getAmount(amount: number): string {
-        return formatTokenAmountBestMatch(amount, baseCoin.metadata)
+        return formatTokenAmountBestMatch(amount, baseCoin?.metadata as TokenMetadata)
```
Comment (VmMad) [inline packages/shared/components/BalanceSummarySection.svelte:21]:
```suggestion
        return baseCoin?.metadata ? formatTokenAmountBestMatch(amount, baseCoin?.metadata) : ''
```

File: packages/shared/components/BalanceSummarySection.svelte [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
     }
 
     function getCurrencyAmount(amount: number): string {
-        return formatCurrency(getMarketAmountFromAssetValue(amount, baseCoin))
+        return formatCurrency(getMarketAmountFromAssetValue(amount, baseCoin as IAsset))
```
Comment (VmMad) [inline packages/shared/components/BalanceSummarySection.svelte:25]:
```suggestion
        return baseCoin ? formatCurrency(getMarketAmountFromAssetValue(amount, baseCoin)) : ''
```

File: packages/shared/components/BalanceSummaryRow.svelte [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        color: 'gray-600',
+        darkColor: 'gray-400',
+        fontSize: '13',
+        fontWeight: FontWeight.normal,
+        lineHeight: '4',
+    }
+</script>
+
+<div class="flex flex-row justify-between flex-grow">
+    <div class={title ? 'flex flex-col space-y-0.5' : ''}>
```
Comment (VmMad) [inline packages/shared/components/BalanceSummaryRow.svelte:28]:
```suggestion
    <div class={title ? 'flex flex-col space-y-0.5' : null}>
```

**Review by VmMad — CHANGES_REQUESTED** [review]

**Review by brancoder — APPROVED** [review]
LGTM

**Review by VmMad — APPROVED** [review]

**Review by begonaalvarezd — APPROVED** [review]
😸

**Pull request merged**